### PR TITLE
fix(providers): ensure multi-part upload is disabled for S3 backends

### PR DESCRIPTION
**Pull request by julio-lopez**

### CODE REVIEW

**Comment by julio-lopez** [comment]
Partially addresses #1909, at least for the desire objective.

File: repo/blob/s3/s3_storage.go [hunk 1]
```diff
@@ -172,7 +172,8 @@ func (s *s3Storage) putBlob(ctx context.Context, b blob.ID, data blob.Bytes, opt
 	}
 
 	uploadInfo, err := s.cli.PutObject(ctx, s.BucketName, s.getObjectNameString(b), data.Reader(), int64(data.Length()), minio.PutObjectOptions{
-		ContentType: "application/x-kopia",
+		ContentType:      "application/x-kopia",
+		DisableMultipart: true,
```
Comment (adowair) [inline repo/blob/s3/s3_storage.go:176]: Can we add a comment explaining why we do this? Along the lines of: `"Kopia already splits snapshot contents into small blobs for uploading, so there is no need for further splitting"`.

**Review by adowair — COMMENTED** [review]
Thanks for putting this up @julio-lopez

File: repo/blob/s3/s3_storage.go (hunk 1 above)
Comment (julio-lopez) [reply repo/blob/s3/s3_storage.go:176]: @adowair Good idea. Please submit a PR with the change. 🙏🏼

**Pull request merged**